### ISSUE TITLE
Invalid PKCS8 encoding for private keys in python-rsa backend

### ISSUE THREAD

**Issue by mattsb42-aws** [mpdavis/python-jose#119]
In the process of investigating the `pycrypto/dome` dependency on the `python-rsa` backend, I discovered that the PKCS8 encoding being performed for RSA private keys by the `python-rsa` backend is incompatible with the other backends.

Digging deeper, I found that the magic prefix[1] being added to the PKCS1 encoded private key[2] is both invalid per the PKCS8 spec[3] and incorrectly sets a static message size.

This results in the serialized keys being unreadable by any other system, and also in the `python-rsa` backend being unable to deserialize any keys written by a correct system.

Current magic prefix:

[Full key from tests, header tweaked to have a valid length](https://lapo.it/asn1js/#MIIJQQIBADANBgkqhkiG9w0BAQEFADCCCSsCAQACggIBALUin0niNH7pESF9_AB5SgdWD6V_D1GDdiXb3zPl4zZi31NTO-DdFZncyF_ebJ3kBjvZAtsTCBPgCJbedmH_yoAWYIAJUvWNFcfA7Ei9h7zo4LgHV4q972C7wMsh4p_5lIrCTqnHBSgoRyo55NLl8v9_Pta9e_3GE6e-m6yqVNSOJDNtDP_3W7ywVo388sPXobn6--GlcK_tMSX7AVa9qGkBcMP1xxs-vUO8hyug28WDuMOKtrCH3AuKU_F0zx6OCWdjO99xGvGux8bWUuet_5oYUWS1OWsp0KcGlb9lPvgi-hLxrfE5TWTpHkb_MM_kbfAe9I86EaVSt-q0fqRypV4TBk-tfb_Ni53k3bKgNuVoti3f3NJ4rrpduAOvmmo9rvUlm8QPS5lbRZ7bzW0Wh1xNUi6Sz6CKfLqaRhdNjc9r95XfIAp001n6vwUPNEMvHtHKEUQARAma4yDMxxIOjJaEQ8uJ2tKUUL-tVaIKkSg1Nq9_1XsxT0A293ImLGY1ga9x6TTpFI067y5hcjhPUOUf6kBpnOgWLX5Oa5-4iH15ZCQGR14QcvhJQbogTPmEpBTO3R_drEiKGdOVeDD9PV3Kace5HcLCcu9krrLfR53fQe1d-WJ1Relu_dZVR53p4QiTs4kZpB-MSy2z5Gkk9irNyBx-7VZbTOjbEZN4zXVTAgMBAAECggIBAK8ftCV4oAx7RWa-KWBD48DIAgSdna_Pi_D6bQf-IPi6CvTCqkezOGkzvj6CCz1z8lr2av5nng2pMmS63HXPGndQKyhe22gwaXhhG5EQPSX1eR4zav3muIMrwzAhqLvGT0kAp5EZq_CxUGyQ4JzOWWuQGK8BL9mhIeuyK0x6ud1vN6zIqCLpgjYhvu00O3oBBomLEO-ORi6xAi2YSikU4Lf0_pNXEpNSyyWsJnuV4CVMPtw_RnXSRHqb2KC_sGf4JztgA8j5z3UO6HNjT3BTF6ZiEH9vfv4OxX5WrX0IZCL_ngumwedQ4XTItc8qdoTocyoOo5--IsVV_h7bNv3DIgD7tDWlplPZ_IOx5l1nOPrkjv37LBoyNwy2CnecZ4a6uGt2TuaCS3jDvVO9fmJHLYNtz0x8DQBq7D2HYxnAZyRHbW12t9WoCOTwBuq4BlGN4kqFqhy4AU8a19OE1lby0IvMG8YewRBdMmrDnmSsXRNiFmIO-cxrI5bsh5Tp6UWuZUn-imerYYfXnMxR2LUuA72AXGfYWVv4PO4ntvEpARccZdG62LfbijWCb7mo1RvJE_NAAu1s4Nqu1pp6MMNyTSVGzbYbOXaislEpTBTCAf7znjxRy6wOqg3wJ7EB4FG_Vh4VK2LB9tFyqakdjo5llNOW1eKUjFvycBt-6PqcdcihAoIBAQDZ8MdHe5LBDLyY3jeaZQ8t3jC4nwSPuki_6OU1w6BTOqycRftFD-9LqLmCatnSPgwkgOnPxLJ-1gDoq2Ne-2nSnAVfG4sALZih5xZwrP5MAGavJTL0D7TTyf76tp-NJDuPIfH8W3hnJsgKhT2eGCksazwOHHWxyLhUrfVsmlHFRZs3UxR1ZUT-9BF3Zlx7fOG8ljQeAh3qvM5qzcTzQIuuZxJFJLO4gUDLbr8ldvml4wzkwTAneuiISIdomatTu5F1MIIdTh_YQmu6K4h8gHYJCnSglqHJmzzxDThE4I62gPcESF8DmEw29V-YTL4tZgh6PMYzI7uLMw9N8SVYu1q7AoIBAQDUxG5QpMRMoSamjGRd1N3X9mK_2bR6rK06-JcAfzDipM5rkq0TsYXhJCiLMW4lfgMU77T0dRUK5am4UyUcMvqcOCjyez-H6tkf2Aar10jQQmpk7epdl3V6pwJNwPvnDhGdpVPbm34HXjdiIfK4HZ9S1njkowlnxMZpKFOtf13usQDHlykEpRykp1_b6MbhNsSDR3lOmQCstZBpqybRIlpyPG_AuLPH08g7VydS2rMrNIhztk89o1IX-CCcG_Oy5OyM3tjKyPqD81Q15ZpG45AeLWMqIkU4_K5PKcAVvx1b9NNqYzi02uJM9ZEN5bGf9wLOM3zYu5lhIWnZhLACH2JJAoIBAQDTt_a_2Ko-ZFMq5mV51ccjNgB6ufBCeCOIW4Wf70Vm1U8uGUX6V3qOM4DT0117ws8k_x8kud71HIyRez3z3aV19h-5vxYPvDvUvJuuJkB8ML-QUkDnnAJ85HSRtqvU_2fkqoNcNrgG7UPUBJBRbwNApYQX6UnkxitcCAqt0FSzoeUhn9H2IcUfMJdvOL-LL0xUWk6TAFdz3KtiUjeMYB3R9UtoZDk7ekUp25JRoPzxTFsQNyTClcIj8uGomfA4TbUG9XLRaT3CZvQkTXowCNOiAMg_4VWWdvqC6ebJ8qRxY2OUg4HaCi-wDDsrxxHRJJgDt9qLf6EHnzi07Rjs1EVVAoIBAQC0seYmIuh7U9kpVM3gSmnlgWA4IsH99SxhisFjMKHpuaF9BmJq-TcED9tG60HqIWyomTMK8WxfhtButF4t5rWjeqZ72GQKIE8pliOESR-TjvQgp1WFCp5A_hkcw6qrfe1D_yaKuTF9PGy4sLAb4Txv86lUM4pHUHxYzmDSVfsGPdi1qRCy2y7KP0NP1g8hMYwPGeJR9-r0wnXU5__dWNmLbvxRpgs4yAmjK88_tHC5XrIL42bEqDGOHbJEIhEDexvSP2fKQIlRCpQX-djeH2FD37P6EoTLcvzuSjzRuy9J61CpZ36_Sa0rQtpf_RSvD-6YBG4g-qG2NdRZYTDBfLnRAoIBAQCCobzhbYqQ9Y6gzqYzqEfbCv5UUeW1VVkH2pjAzdQMJoyW1R0vIYbWFDP-LIdqddj-kYKDvHzg39bxHFhYd8cTWRNaTwj2iAg_YuVPjUbz89rwvdNB3K2i0a1BWkc8IajjpJ2CUgaxs1vgsd2EnmjgoJysiPAeYMOAtBtIi9XUrM9m0dTuBjTlX090eo6GRFwExaPynNi9GALwKQTVGL2NJG4yfyX0zudOtErFn7X-IsN464Up_UcE02Hav5BKxhVrwxiZ9jIroTHtqJzX1cyBkZnVMR8ItbpZLKQJ_35mO39IWabJA8HB8mZmymbpPjVPxSfCAHJr5Pcu5tuZ0knP)

```
30  # sequence
8204BD  # DER-encoded sequence contents length of 1213 bytes -- INCORRECT STATIC LENGTH
  020100  # integer: 0 -- Version
  300D  # sequence containing 13 bytes -- PrivateKeyAlgorithmIdentifier
    06092A864886F70D010101  # OID -- rsaEncryption
    0500  # NULL -- parameters
  <PKCS1 key added>
```

Prefix should be:

[Full key from tests](https://lapo.it/asn1js/#MIIJRQIBADANBgkqhkiG9w0BAQEFAASCCS8wggkrAgEAAoICAQC1Ip9J4jR-6REhffwAeUoHVg-lfw9Rg3Yl298z5eM2Yt9TUzvg3RWZ3Mhf3myd5AY72QLbEwgT4AiW3nZh_8qAFmCACVL1jRXHwOxIvYe86OC4B1eKve9gu8DLIeKf-ZSKwk6pxwUoKEcqOeTS5fL_fz7WvXv9xhOnvpusqlTUjiQzbQz_91u8sFaN_PLD16G5-vvhpXCv7TEl-wFWvahpAXDD9ccbPr1DvIcroNvFg7jDirawh9wLilPxdM8ejglnYzvfcRrxrsfG1lLnrf-aGFFktTlrKdCnBpW_ZT74IvoS8a3xOU1k6R5G_zDP5G3wHvSPOhGlUrfqtH6kcqVeEwZPrX2_zYud5N2yoDblaLYt39zSeK66XbgDr5pqPa71JZvED0uZW0We281tFodcTVIuks-giny6mkYXTY3Pa_eV3yAKdNNZ-r8FDzRDLx7RyhFEAEQJmuMgzMcSDoyWhEPLidrSlFC_rVWiCpEoNTavf9V7MU9ANvdyJixmNYGvcek06RSNOu8uYXI4T1DlH-pAaZzoFi1-TmufuIh9eWQkBkdeEHL4SUG6IEz5hKQUzt0f3axIihnTlXgw_T1dymnHuR3CwnLvZK6y30ed30HtXflidUXpbv3WVUed6eEIk7OJGaQfjEsts-RpJPYqzcgcfu1WW0zo2xGTeM11UwIDAQABAoICAQCvH7QleKAMe0VmvilgQ-PAyAIEnZ2vz4vw-m0H_iD4ugr0wqpHszhpM74-ggs9c_Ja9mr-Z54NqTJkutx1zxp3UCsoXttoMGl4YRuRED0l9XkeM2r95riDK8MwIai7xk9JAKeRGavwsVBskOCczllrkBivAS_ZoSHrsitMerndbzesyKgi6YI2Ib7tNDt6AQaJixDvjkYusQItmEopFOC39P6TVxKTUsslrCZ7leAlTD7cP0Z10kR6m9igv7Bn-Cc7YAPI-c91DuhzY09wUxemYhB_b37-DsV-Vq19CGQi_54LpsHnUOF0yLXPKnaE6HMqDqOfviLFVf4e2zb9wyIA-7Q1paZT2fyDseZdZzj65I79-ywaMjcMtgp3nGeGurhrdk7mgkt4w71TvX5iRy2Dbc9MfA0Aauw9h2MZwGckR21tdrfVqAjk8AbquAZRjeJKhaocuAFPGtfThNZW8tCLzBvGHsEQXTJqw55krF0TYhZiDvnMayOW7IeU6elFrmVJ_opnq2GH15zMUdi1LgO9gFxn2Flb-DzuJ7bxKQEXHGXRuti324o1gm-5qNUbyRPzQALtbODartaaejDDck0lRs22Gzl2orJRKUwUwgH-8548UcusDqoN8CexAeBRv1YeFStiwfbRcqmpHY6OZZTTltXilIxb8nAbfuj6nHXIoQKCAQEA2fDHR3uSwQy8mN43mmUPLd4wuJ8Ej7pIv-jlNcOgUzqsnEX7RQ_vS6i5gmrZ0j4MJIDpz8SyftYA6KtjXvtp0pwFXxuLAC2YoecWcKz-TABmryUy9A-008n--rafjSQ7jyHx_Ft4ZybICoU9nhgpLGs8Dhx1sci4VK31bJpRxUWbN1MUdWVE_vQRd2Zce3zhvJY0HgId6rzOas3E80CLrmcSRSSzuIFAy26_JXb5peMM5MEwJ3roiEiHaJmrU7uRdTCCHU4f2EJruiuIfIB2CQp0oJahyZs88Q04ROCOtoD3BEhfA5hMNvVfmEy-LWYIejzGMyO7izMPTfElWLtauwKCAQEA1MRuUKTETKEmpoxkXdTd1_Ziv9m0eqytOviXAH8w4qTOa5KtE7GF4SQoizFuJX4DFO-09HUVCuWpuFMlHDL6nDgo8ns_h-rZH9gGq9dI0EJqZO3qXZd1eqcCTcD75w4RnaVT25t-B143YiHyuB2fUtZ45KMJZ8TGaShTrX9d7rEAx5cpBKUcpKdf2-jG4TbEg0d5TpkArLWQaasm0SJacjxvwLizx9PIO1cnUtqzKzSIc7ZPPaNSF_ggnBvzsuTsjN7Yysj6g_NUNeWaRuOQHi1jKiJFOPyuTynAFb8dW_TTamM4tNriTPWRDeWxn_cCzjN82LuZYSFp2YSwAh9iSQKCAQEA07f2v9iqPmRTKuZledXHIzYAernwQngjiFuFn-9FZtVPLhlF-ld6jjOA09Nde8LPJP8fJLne9RyMkXs9892ldfYfub8WD7w71LybriZAfDC_kFJA55wCfOR0kbar1P9n5KqDXDa4Bu1D1ASQUW8DQKWEF-lJ5MYrXAgKrdBUs6HlIZ_R9iHFHzCXbzi_iy9MVFpOkwBXc9yrYlI3jGAd0fVLaGQ5O3pFKduSUaD88UxbEDckwpXCI_LhqJnwOE21BvVy0Wk9wmb0JE16MAjTogDIP-FVlnb6gunmyfKkcWNjlIOB2govsAw7K8cR0SSYA7fai3-hB584tO0Y7NRFVQKCAQEAtLHmJiLoe1PZKVTN4Epp5YFgOCLB_fUsYYrBYzCh6bmhfQZiavk3BA_bRutB6iFsqJkzCvFsX4bQbrReLea1o3qme9hkCiBPKZYjhEkfk470IKdVhQqeQP4ZHMOqq33tQ_8mirkxfTxsuLCwG-E8b_OpVDOKR1B8WM5g0lX7Bj3YtakQstsuyj9DT9YPITGMDxniUffq9MJ11Of_3VjZi278UaYLOMgJoyvPP7RwuV6yC-NmxKgxjh2yRCIRA3sb0j9nykCJUQqUF_nY3h9hQ9-z-hKEy3L87ko80bsvSetQqWd-v0mtK0LaX_0Urw_umARuIPqhtjXUWWEwwXy50QKCAQEAgqG84W2KkPWOoM6mM6hH2wr-VFHltVVZB9qYwM3UDCaMltUdLyGG1hQz_iyHanXY_pGCg7x84N_W8RxYWHfHE1kTWk8I9ogIP2LlT41G8_Pa8L3TQdytotGtQVpHPCGo46SdglIGsbNb4LHdhJ5o4KCcrIjwHmDDgLQbSIvV1KzPZtHU7gY05V9PdHqOhkRcBMWj8pzYvRgC8CkE1Ri9jSRuMn8l9M7nTrRKxZ-1_iLDeOuFKf1HBNNh2r-QSsYVa8MYmfYyK6Ex7aic19XMgZGZ1TEfCLW6WSykCf9-Zjt_SFmmyQPBwfJmZspm6T41T8UnwgBya-T3LubbmdJJzw)

```
30  # sequence
820945  # DER-encoded sequence contents length of 2373 bytes -- LENGTH MUST BE SET PER KEY
  02 01 00  # integer: 0 -- Version
  30  # sequence
  0D  # DER-encoded sequence contents length of 13 bytes -- PrivateKeyAlgorithmIdentifier
    06092A864886F70D010101  # OID -- rsaEncryption
    0500  # NULL -- parameters
  04 # Octet String
  82092F  # DER-encoded octet string length of 2351 bytes -- LENGTH MUST BE SET PER KEY
    <PKCS1 key added>
```

[1] https://github.com/mpdavis/python-jose/blob/master/jose/backends/rsa_backend.py#L15
[2] https://github.com/mpdavis/python-jose/blob/master/jose/backends/rsa_backend.py#L186
[3] https://tools.ietf.org/html/rfc5208#page-5

**Comment by mattsb42-aws** [comment]
Addressed in #120. I put it on top of `backend-explicit-tests` because it's much more difficult to actually isolate and test this behavior before that branch.

**Comment by mattsb42-aws** [comment]
This is resolved with #129
